app/src/main/java/com/eis/geoCalendar/app/GenericTimedEventManager.java

```diff
@@ -0,0 +1,81 @@
+package com.eis.geoCalendar.app;
+
+import com.eis.geoCalendar.events.EventDatabase;
+import com.eis.geoCalendar.timedEvents.DateTime;
+import com.eis.geoCalendar.timedEvents.TimedEvent;
+import com.eis.geoCalendar.timedEvents.TimedEventManager;
+
+import java.util.ArrayList;
+
+/**
+ * This class manages all scheduled events.
+ *
+ * @param <E> The generic timed event.
+ * @author Francesco Bau' helped by Luca Crema.
+ * @version 1.0
+ * @see com.eis.geoCalendar.app.GenericEventManager
+ * @see com.eis.geoCalendar.timedEvents.TimedEvent
+ * @see com.eis.geoCalendar.timedEvents.TimedEventManager
+ * @since 23/12/2019
+ */
+public class GenericTimedEventManager<E extends TimedEvent> extends GenericEventManager<E> implements TimedEventManager<E> {
+
+    /**
+     * Main constructor, needed to acquire the database of the events.
+     *
+     * @param database The database of the events.
+     */
+    public GenericTimedEventManager(EventDatabase<E> database) {
+        super(database);
+    }
+
+    /**
+     * Searches for all events that have a time earlier than the parameter time
+     *
+     * @param time The time upper bound.
+     * @return An {@link ArrayList} of events with time smaller than {@code time} if there is any, an empty array otherwise.
+     */
+    @Override
+    public ArrayList<E> getEventsBeforeTime(DateTime time) {
+        ArrayList<E> eventsList = new ArrayList<>();
+        for (E event : database.getSavedEvents()) {
+            if (event.getTime().compareTo(time) < 0)
```

Comment:
I think also this 0 is hardwiring. 
I'd creat an private final int offsetTime=0 object or with a similar name

Reply:
Personally, I don't know the reason to have a constant OFFSET_TIME = 0...
I think the 0 is the only neutral magic number which may not be considered hardwired...
It's part of the compareTo method, implementation of Comparable interface, which is implemented also in the main library, such as String, Integer, Double, ...
Considering 2 Integers X and Y, as an example...
If X.compareTo(Y)>0 it means that X>Y (e.g. if X=5 and Y=0, it returns 5)
If X.compareTo(Y)=0 it means that X=Y (e.g. if X=5 and Y=5, it returns 0)
If X.compareTo(Y)<0 it means that X<Y (e.g. if X=0 and Y=5, it returns -5)
Also, I got inspired by methods in the Java library....
Watch these methods:
```
public boolean regionMatches(int toffset, String other, int ooffset,
            int len) {
        int to = toffset;
        int po = ooffset;
        // Note: toffset, ooffset, or len might be near -1>>>1.
        if ((ooffset < 0) || (toffset < 0)
                || (toffset > (long)length() - len)
                || (ooffset > (long)other.length() - len)) {
            return false;
        }
        while (len-- > 0) {
            if (charAt(to++) != other.charAt(po++)) {
                return false;
            }
        }
        return true;
    }
```
This is just an example. In general, I consider something hardwired if and only if that value might change in time, and if that value is "connected" in other classes such as hashCodes, or Strings in databases (if someone asks to change the title, and that entity is on many databases, it's a total chaos).
So, in general, if it's something "connected" in other classes, it's hardwiring. In the other cases, it might not be the case.
Thank you for bringing this out, talking with you helped me reflect on my hardwiring doubts...

Reply:
Yeah, I wouldn't consider this a hardwiring as well, given how you use it, since making it a const variable and then updating it might work in a function but break the other ones.